streamx-console/streamx-console-service/src/main/java/com/streamxhub/streamx/console/core/service/impl/AlertServiceImpl.java

```diff
@@ -179,4 +187,49 @@ private MailTemplate getMailTemplate(Application application) {
         return template;
     }
 
+    /**
+     * dingding  alert
+     * @param application
+     * @throws Exception
+     */
+    private void sendDing(Application application){
+        try {
+            if (dingdingProperties.isEnabled() && application != null) {
+                StringBuffer stringBuffer = new StringBuffer("StreamX >>>>>>>>> ID: ");
+                stringBuffer.append(application.getId());
+                stringBuffer.append(",JOB NAME: ");
+                stringBuffer.append(application.getJobName());
+                stringBuffer.append(" FAIL,SavePointed: ");
+                stringBuffer.append(application.getSavePointed());
+                stringBuffer.append(",SavePoint: ");
+                stringBuffer.append(application.getSavePoint());
```

Comment:
stringBuffer.append(application.getId())
               .append(",JOB NAME: ")
               .append(application.getJobName())
               .append(" FAIL,SavePointed: ")
               .append(application.getSavePointed())
               .append(",SavePoint: ")
               .append(application.getSavePoint());

It will be easier and more readable to write like this

Reply:
I have use  String.format  replace StringBuffer.append